### PR TITLE
scripts/build_utils: add copr repo for latest gcc-toolset-9

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
being tested at https://shaman.ceph.com/builds/ceph/wip-kefu2-testing-2020-08-15-1257/

File: scripts/build_utils.sh [hunk 1]
```diff
@@ -1088,6 +1088,83 @@ maybe_reset_ci_container() {
...
+    *)
+        DIST=unknown
+        DISTRO=unknown
+        ;;
+    esac
+
+    echo $DIST
+}
+
+setup_rpm_build_deps() {
```
Comment (rzarzynski) [inline scripts/build_utils.sh:1136]: This commit significantly deduplicates the code. Assuming it brings no other changes.

File: scripts/build_utils.sh [hunk 2]
```diff
@@ -1155,6 +1155,9 @@ setup_rpm_build_deps() {
     if [ "$FLAVOR" = "crimson" ]; then
         # enable more build depends required by crimson
         sed -i -e 's/%bcond_with seastar/%bcond_without seastar/g' $DIR/ceph.spec
+        # use gcc-toolset-9 for the better support of C++17 from GCC
+        # see also https://bugzilla.redhat.com/show_bug.cgi?id=1853900
+        $SUDO dnf copr enable -y tchaikov/gcc-toolset-9 centos-stream-x86_64
```
Comment (rzarzynski) [inline scripts/build_utils.sh:1160]: OK, only for `"$FLAVOR" = "crimson"`.

**Comment by tchaikov** [comment]
being tested at https://shaman.ceph.com/builds/ceph/wip-kefu2-testing-2020-08-15-1631/ae8bf8336c599a7b64ecc0da9c915d9457283511/crimson/223250/

**Pull request merged**